nulib/dc-nextjs
# Wire up AI chat feedback form

## What does this do?

This work wires up the Feedback form to the DC API `*/chat/feedback` endpoint.

Two submissions take place:

![image](https://github.com/user-attachments/assets/4edb16fd-51de-44d0-8576-c7adfa19d8d6)

1) On initial click of 👍 or 👎 , the payload carrying a `sentiment` of **positive** or **negative**, along with the question `context` is submitted, creating a file in the S3 bucket for unique `context.ref`

After this click event, the thumbs up/down buttons are disabled and the form expands, allowing the user to complete with more verbose feedback.

![image](https://github.com/user-attachments/assets/f7d624cf-7075-4a45-9185-94c2ff7567ac)

2) If this second submission is completed, the payload is appended to with the `options`, `text`, and user `email` fields, and submitted as the same `context.ref`, overwriting the relative file in the S3 bucket.


## How to review

The form should behave as follows:

1) Complete a AI query and noted the "Was this answer helpful? 👍 👎" line below the streamed Chat/AI response
2) Select a sentiment option by click the appropriate thumb icon,
  a) a `POST` occurs to the DC API endpoint
  b) a file is written in the relative S3 bucket
  c) the form is expanded
  d) sentiment icons are disabled
3) Complete the remainder of the form
4) Submit again using the UI button
  a) a `POST` occurs to the DC API endpoint
  b) the relative `ref` file is overwritten in the relative S3 bucket
  c) the goes away
  d) a thank you message is rendered
